File: Leetcode/209-Minimum-Size-Subarray-Sum/209-Minimum-Size-Subarray-Sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int i = 0, j = 0, sum = 0, n = nums.size(), minLength = n + 1;
        while (j <= n) {
            if (j < n && sum < target) {
                sum += nums[j];
                j++;
            } else if (i < n && sum >= target) {
                minLength = min(minLength, j - i);
                sum -= nums[i];
                i++;
            } else {
                j++;
            }
        }

        return (minLength <= n) ? minLength : 0;
    }
};

File: Leetcode/209-Minimum-Size-Subarray-Sum/209-Minimum-Size-Subarray-Sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) { 
        int currSum = nums[0], minLen = nums.length + 1, left = 0, right = 0;

        while(right < nums.length && left <= right) {
            if (currSum < target) {
                right++;

                if (right < nums.length) {
                    currSum += nums[right];
                }
            } else {
                minLen = Math.min(minLen, right - left + 1);
                currSum -= nums[left++];
            } 
        }

        return (minLen == nums.length + 1) ? 0 : minLen;
    }
}